compiler/compiler.go

```diff
@@ -2182,21 +2182,31 @@ func (c *Compiler) parseExpr(frame *Frame, expr ssa.Value) (llvm.Value, error) {
 
 		// Bounds checking.
 		if !frame.fn.IsNoBounds() {
-			if sliceLen.Type().IntTypeWidth() < c.uintptrType.IntTypeWidth() {
+			checkFunc := "sliceBoundsCheckMake"
+			biggestInt := c.uintptrType
+			biggestIntWidth := biggestInt.IntTypeWidth()
+			if sliceLen.Type().IntTypeWidth() > biggestIntWidth || sliceCap.Type().IntTypeWidth() > biggestIntWidth {
+				// System that is less than 64bit, meaning that the slice make
+				// params are bigger than uintptr.
+				checkFunc = "sliceBoundsCheckMake64"
```

Comment:
but you are still truncating the capacity below on line 2215

Reply:
Good point, this is an issue, especially when the slice contains elements bigger than a byte. I'll have to think about a proper fix for this.

Reply:
For reference, this is where the main Go compiler checks slice bounds:
https://github.com/golang/go/blob/master/src/runtime/slice.go#L34